libs/renterd-types/src/types.ts

```diff
@@ -156,6 +158,50 @@ export type S3AuthenticationSettings = {
   }
 }
 
+export type Pin = {
+  pinned: boolean
+  value: number
+}
+
+export type GougingSettingsPins = {
+  maxStorage: Pin
+  maxDownload: Pin
+  maxUpload: Pin
+  maxRPCPrice: Pin
+}
+
+export type AutopilotPins = {
+  allowance: Pin
+}
+
+// PricePinSettings holds the configuration for pinning certain settings to
+// a specific currency (e.g., USD). It uses a Forex API to fetch the current
+// exchange rate, allowing users to set prices in USD instead of SC.
+export type PricePinSettings = {
+  // Enabled can be used to either enable or temporarily disable price
+  // pinning. If enabled, both the currency and the Forex endpoint URL
+  // must be valid.
+  enabled: boolean
+
+  // Currency is the external three-letter currency code.
+  currency?: string
```

Comment:
`currency`, `threshold` and `forexEndpointURL` are mandatory right now

Reply:
👍🏼 @ChrisSchinnerl I'll update that. This is what I get initially on a fresh node, looks like it returns a default object for gouging but not for autopilot?
```
{
	"currency": "",
	"enabled": false,
	"forexEndpointURL": "",
	"gougingSettingsPins": {
		"maxDownload": {
			"pinned": false,
			"value": 0
		},
		"maxRPCPrice": {
			"pinned": false,
			"value": 0
		},
		"maxStorage": {
			"pinned": false,
			"value": 0
		},
		"maxUpload": {
			"pinned": false,
			"value": 0
		}
	},
	"threshold": 0
}
```

Reply:
Indeed. But probably changing that to not omit anything and return all fields that can be set https://github.com/SiaFoundation/renterd/pull/1399